test/test_basebackup_operation.py

```diff
@@ -125,5 +126,74 @@ def stream_handler(_stream):
         stream_handler=stream_handler,
         temp_dir=mysql_master.base_dir,
     )
-    with pytest.raises(Exception, match="^xtrabackup failed with code 13$"):
+    # we're opening a connection to check the version on mysql >= 8.0.29 below we're failing with the first message
+    with pytest.raises(Exception, match=r"(^xtrabackup failed with code 13$|^mysql_cursor\(\) missing 3 required keyword)"):
         op.create_backup()
+
+
+def test_backup_with_non_optimized_tables(mysql_master) -> None:
+    with myhoard_util.mysql_cursor(**mysql_master.connect_options) as cursor:
+        version = myhoard_util.get_mysql_version(cursor)
+        if version < "8.0.29":
+            raise SkipTest("DB version doesn't support OPTIMIZE TABLE")
```

Comment:
nit. OPTIMIZE TABLE is supported on older versions, it's more that it's not required in this  situation.